### PR TITLE
collapse contributors with no answers in course detail pages

### LINKED ISSUE #555: collapse contributors with no answers in course detail pages
Contributors who didn't get any answers should be collapsed on the results pages, so that the empty answer lines are not shown.
This should also happen if there are answers in the database, but none of them can be seen by the current user.

### PR DESCRIPTION
**Pull request by schumannd**
Fixes #555

### CODE REVIEW

**Comment by schumannd** [comment]
Fixed up and ready for review.

File: evap/results/views.py [hunk 1]
```diff
@@ -80,14 +79,21 @@ def course_detail(request, semester_id, course_id):
...
         else:
-            contributor_sections.setdefault(section.contributor, []).append(section)
+            contributor_sections.setdefault(section.contributor,
+                                            {'total_votes': 0, 'sections': []})['sections'].append(section)
+
+            # Sum up all Sections for this contributor.
+            # If section is not a RatingResult:
+            # Add 1 as we assume it is a TextResult or something similar that should be displayed.
+            contributor_sections[section.contributor]['total_votes'] +=\
+                sum([s.total_count if isinstance(s, RatingResult) else 1 for s in section[3]])
```
Comment (janno42) [inline evap/results/views.py]: `ResultSection` is a namedtuple, defined in `evaluations/tools.py`. please use the name instead of `section[3]`

**Comment by janno42** [comment]
the arrows in the panel headings are not pointing down when the panel is initially rendered in an open state

**Comment by janno42** [comment]
please add a message "There are no results for this contributor" with a warning sign (like the message when only a few participants voted for a question in `results/templates/results_course_detail_questionnaires.html`)

File: evap/results/templates/results_course_detail.html [hunk 2]
```diff
@@ -103,19 +103,26 @@
                 <span class="panel-title">{% trans "Contributors" %}</span>
             </div>
             <div class="panel-body">
-                {% for contributor,sections in contributor_sections.items %}
-                    {% regroup sections by questionnaire as sections_grouped %}
+                {% for contributor, contributor_data in contributor_sections.items %}
```
Comment (janno42) [inline evap/results/templates/results_course_detail.html]: please add a `<div class="panel-group">` around the panels created in this for loop

**Comment by schumannd** [comment]
done

**Comment by janno42** [comment]
it's broken:
broken panel

- please make the message not a tooltip but always visible like the others.
- please move the `<div class="panel-group">` *around* the for loop

**Comment by schumannd** [comment]
Sorry about that. Should be working now.

File: evap/results/templates/results_course_detail.html [hunk 3]
```diff
@@ -103,22 +103,50 @@
...
                         <div class="panel-heading">
                             <span class="panel-title">
-                                {{ contributor.full_name }}
-                                {% if sections_grouped.0.list.0.label %} &ndash; <i>{{ sections_grouped.0.list.0.label }}</i>{% endif %}
+
+                        {% comment %} Logic for collapsing Contributor Sections with no votes. {% endcomment %}
+                        {% if contributor_data.total_votes == 0 %}
+
+                                <div class="inline-headers">
+                                    <p class="pull-right participants-warning questionnaire-warning">
```
Comment (janno42) [inline evap/results/templates/results_course_detail.html]: please remove the `questionnaire-warning`

File: evap/results/templates/results_course_detail.html [hunk 4]
```diff
@@ -103,22 +103,50 @@
...
+                                        <span class="glyphicon glyphicon-warning-sign"></span>
+                                        {% trans "There are no results for this contributor" %}
+                                    </p>
+                                </div>
+
+                                 <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#{{ contributor.id }}">
+                                     {{ contributor.full_name }}{% if sections_grouped.0.list.0.label %} &ndash; <i>{{ sections_grouped.0.list.0.label }}</i>{% endif %}
+                                     <span data-toggle="tooltip" data-placement="right" class="glyphicon glyphicon-info-sign"
```
Comment (janno42) [inline evap/results/templates/results_course_detail.html]: please remove the info sign and tooltip

File: evap/results/templates/results_course_detail.html [hunk 5]
```diff
@@ -103,22 +103,50 @@
...
                             </span>
                         </div>
-                        <div class="panel-body">
-                            {% for group in sections_grouped %}
-                                {% include "results_course_detail_questionnaires.html" %}
-                            {% endfor %}
+                        <div id="{{ contributor.id }}" class="panel-collapse collapse">
+
+                        {% else %}
+
```
Comment (janno42) [inline evap/results/templates/results_course_detail.html]: please fix all the whitespace and indentations around here

**Comment by janno42** [comment]
can you remove the message "Only a few participants answered these questions." for all questionnaires of a contributor who already has the message "There are no results for this contributor"?

**Comment by schumannd** [comment]
Done

File: evap/results/templates/results_course_detail.html [hunk 6]
```diff
@@ -103,22 +103,39 @@
...
-                    {% regroup sections by questionnaire as sections_grouped %}
+                <div class="panel-group">
+                {% for contributor, contributor_data in contributor_sections.items %}
+                    {% regroup contributor_data.sections by questionnaire as sections_grouped %}
                     <div class="panel panel-default">
                         <div class="panel-heading">
                             <span class="panel-title">
-                                {{ contributor.full_name }}
-                                {% if sections_grouped.0.list.0.label %} &ndash; <i>{{ sections_grouped.0.list.0.label }}</i>{% endif %}
+                        {% comment %} Logic for collapsing Contributor Sections with no votes. {% endcomment %}
```
Comment (janno42) [inline evap/results/templates/results_course_detail.html]: sorry, the indentation here is still wrong

**Pull request merged**